runtime/common/src/main/java/org/apache/nemo/runtime/common/metric/TaskMetric.java

```diff
@@ -106,6 +116,35 @@ private void setTaskDuration(final long taskDuration) {
     this.taskDuration = taskDuration;
   }
 
+  /**
+   * Method related to stream metric.
+   *
+   * @return the streamMetrics
+   */
+  public final Map<String, List<StreamMetric>> getStreamMetric() {
+    return streamMetrics;
+  }
+
+  private void setStreamMetric(final Map<String, StreamMetric> streamMetricMap) {
+    for (String sourceVertexId : streamMetricMap.keySet()) {
+      StreamMetric streamMetric = streamMetricMap.get(sourceVertexId);
+      streamMetrics.putIfAbsent(sourceVertexId, new LinkedList<>());
+      streamMetrics.get(sourceVertexId).add(streamMetric);
+    }
+  }
+
+  /**
+   * Method related to latency.
+   */
+  public final Map<String, List<LatencyMetric>> getLatencymarks() {
```

Comment:
Where is it used?

Reply:
It is needed to storing json file, too. If it doesn't exist, metrics are not recorded in json file.

Reply:
It doesn't look like this is used for storing json file.